src/preamble.js

```diff
@@ -514,6 +514,15 @@ function initRuntime() {
   assert(!runtimeInitialized);
 #endif
   runtimeInitialized = true;
+#if WASM_BACKEND
+#if STACK_OVERFLOW_CHECK >= 2
+  Module['___set_stack_limit'](STACK_MAX);
+#endif
+  // Emscripten adds static data during JS compilation (see makeStaticAlloc).
+  // This effects the stack layout which means before we run any code we need
+  // to set stack pointer on the wasm side.
+  Module['stackRestore'](STACKTOP);
```

Comment:
this will increase code size. can we maybe instead avoid the allocations that move STACKTOP during startup? who are they?

Reply:
You are correct, I think I can do it with `--post-emscripten` instead.

There are bunch of allocations that move this, basically all calls to `makeStaticAlloc` that happen during JS compilation.

Reply:
I see. Yeah, sounds like post-emscripten might be the best place. It happens after both the JS compiler and wasm-emscripten-finalize, so we should have all the info at that time.